inc/azure_c_util/interlocked_if.h

```diff
@@ -0,0 +1,156 @@
+#ifndef INTERLOCKEDIFIF_H
+#define INTERLOCKEDIFIF_H
+
+#include <stdint.h>
+
+long InterlockedIFAdd( 
+    long volatile *Addend,
```

Comment:
>Addend [](start = 19, length = 6)

I just thought of this as well, but we generally use all lowercase names with underscores.

I think the function names are fine, but I'd at least clean up the variable names to just be all lowercase (e.g. you have both `Value` and `value`).